activerecord/lib/active_record/connection_adapters/abstract/database_statements.rb

```diff
@@ -396,6 +396,62 @@ def sanitize_limit(limit)
         end
       end
 
+      def build_insert_all_sql(table_name, keys, values_list, updatable_columns:, returning:, on_duplicate:, conflict_target:)
```

Comment:
Feels like we should find a way to extract this builder into it's own object as well. A 30-line method is a wee bit much.